--- C:/Works/PclW/Libs/cgal/include/CGAL/GMP/Gmpzf_type.h.old
+++ C:/Works/PclW/Libs/cgal/include/CGAL/GMP/Gmpzf_type.h
@@ -82,7 +82,7 @@
 
   // exponent type
   // --------------------
-  typedef long Exponent; // may overflow, but if it does, the mantissa is
+  typedef mpir_si Exponent; // may overflow, but if it does, the mantissa is
                          // potentially too large to be useful, anyway;
                          // still, repeated squaring of a power of two
                          // quickly brings this type to its limits...
@@ -420,7 +420,7 @@
   // get surrounding interval of the form [l * 2 ^ k, u * 2^ k]
   // first get mantissa in the form l*2^k, with 0.5 <= d < 1;
   // truncation is guaranteed to go towards zero
-  long k = 0;
+  Exponent k = 0;
   double l = mpz_get_d_2exp (&k, man());
   // l = +/- 0.1*...*
   //           ------
